### PR TITLE
[Internal] Routing: Fixes using ContainerProperties.PartitionKeyPaths for MultiHash collections.

### PR DESCRIPTION
**Pull request by SrinikhilReddy**
# Avoid using ContainerProperties.PartitionKeyPath for MultiHash collections.

## Description

We had introduced [this](https://github.com/Azure/azure-cosmos-dotnet-v3/commit/9a7537318fad949cc6281b359209eb5dea66732f#diff-f85441f11c855d2d9778bc13ab38a470127fa3dfe47240bea2d9659bbcca81b5L276) change earlier for MultiHash collections to avoid using `ContainerProperties.PartitionKeyPath` but instead use `ContainerProperties.PartitionKeyPaths` since the collection is expected to have multiple partition key paths. 

This change avoids and replaces the usage of `PartitionKeyPath` for MultiHash collections with `PartitionKeyPaths` in the existing code.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Database/DatabaseCore.cs [hunk 1]
```diff
@@ -194,17 +196,35 @@ internal abstract class DatabaseCore : DatabaseInternal
                     ContainerResponse retrivedContainerResponse = this.ClientContext.ResponseFactory.CreateContainerResponse(
                         container,
                         readResponse);
-                    if (!retrivedContainerResponse.Resource.PartitionKeyPath.Equals(containerProperties.PartitionKeyPath))
+                    if (containerProperties.PartitionKey.Kind != Documents.PartitionKind.MultiHash)
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Database/DatabaseCore.cs:199]: Please add tests

Comment (SrinikhilReddy) [reply Microsoft.Azure.Cosmos/src/Resource/Database/DatabaseCore.cs:199]: Added them.

**Review by SrinikhilReddy — COMMENTED** [review]
Added tests.

**Comment by SrinikhilReddy** [comment]
image

**Review by ealsur — APPROVED** [review]

**Review by j82w — APPROVED** [review]

**Pull request merged**